### PR TITLE
Feat(ticketing) Remove Osticket

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
Cleaned up osTicket-related code, tests, config settings, and documentation due to the transition to Freshdesk

### Changed

- clean up `osticket` code and documentation from the `TicketHandler`, `OrdersAPI`, etc.

### CODE REVIEW

File: cg/clients/freshdesk/freshdesk_client.py [hunk 1]
```diff
@@ -60,3 +58,27 @@ def _configure_retries(session: Session) -> None:
...
+        files = prepare_attachments(attachments) if attachments else None
+        data = {
+            "body": reply["body"],
+            "from_email": from_email,
+        }
+        response = self.session.post(url=url, data=data, files=files)
+        if response.status_code == HTTPStatus.OK:
+            LOG.info("Successfully replied to ticket %s", ticket_id)
+        else:
+            LOG.error(
```
Comment (seallard) [inline cg/clients/freshdesk/freshdesk_client.py:78]: Doesn't this belong in the error handler?

File: cg/clients/freshdesk/utils.py [hunk 2]
```diff
@@ -25,7 +25,16 @@ def wrapper(*args, **kwargs):
         try:
             return func(*args, **kwargs)
         except (MissingSchema, HTTPError, ConnectionError) as error:
-            LOG.error(f"Request to Freshdesk failed: {error}")
+            if isinstance(error, HTTPError) and error.response is not None:
```
Comment (seallard) [inline cg/clients/freshdesk/utils.py:28]: If there is logic specifically for `HTTPError`, just move that that to a separate line.

File: cg/clients/freshdesk/freshdesk_client.py [hunk 3]
```diff
@@ -60,3 +58,27 @@ def _configure_retries(session: Session) -> None:
...
+        url = f"{self.base_url}{EndPoints.TICKETS}/{ticket_id}/reply"
+
+        files = prepare_attachments(attachments) if attachments else None
+        data = {
+            "body": reply["body"],
+            "from_email": from_email,
+        }
+        response = self.session.post(url=url, data=data, files=files)
+        if response.status_code == HTTPStatus.OK:
+            LOG.info("Successfully replied to ticket %s", ticket_id)
```
Comment (islean) [inline cg/clients/freshdesk/freshdesk_client.py:76]:
```suggestion
            LOG.info(f"Successfully replied to ticket {ticket_id}")
```

File: cg/meta/orders/ticket_handler.py [hunk 4]
```diff
@@ -188,24 +195,26 @@ def connect_to_ticket(
...
-        )
-        email_form.submit()
-        temp_dir.cleanup()
+
+        with TemporaryDirectory() as temp_dir:
+            attachments: Path = self.create_attachment_file(order=order, temp_dir=temp_dir)
+
+            reply_payload = {"body": message, "attachments": []}
+
+            LOG.info("Reply payload: %s", reply_payload)
```
Comment (islean) [inline cg/meta/orders/ticket_handler.py:211]:
```suggestion
            LOG.info(f"Reply payload: {reply_payload}")
```

**Review by islean — COMMENTED** [review]
This PR is quite sparsely type hinted, which is strongly encouraged (https://atlas.scilifelab.se/infrastructure/development/coding_guidelines/#type-hints). Would you consider adding some type hints to make the code base homogeneous?

File: cg/clients/freshdesk/freshdesk_client.py [hunk 5]
```diff
@@ -60,3 +58,27 @@ def _configure_retries(session: Session) -> None:
...
+        else:
+            LOG.error(
+                "Failed to reply to ticket %s. Status code: %s, Response: %s",
+                ticket_id,
+                response.status_code,
+                response.text,
+            )
```
Comment (Vince-janv) [inline cg/clients/freshdesk/freshdesk_client.py:83]: I think an is-statement only for logging makes the code more confusing. I think the error logging belongs in the error-handler as SA suggests

File: cg/clients/freshdesk/models.py [hunk 6]
```diff
@@ -19,9 +18,27 @@ class TicketCreate(BaseModel):
...
+            if isinstance(value, list) and field == "tags":
+                multipart_data.extend([("tags[]", tag) for tag in value])
+            elif isinstance(value, dict) and field == "custom_fields":
+                multipart_data.extend(
+                    [(f"custom_fields[{key}]", val) for key, val in value.items()]
+                )
+            else:
+                multipart_data.append((field, value))
+
+        return multipart_data
```
Comment (Vince-janv) [inline cg/clients/freshdesk/models.py:37]: We could consider using a [custom serialiser](https://docs.pydantic.dev/latest/api/functional_serializers/#pydantic.functional_serializers.model_serializer) to make model_dump() create multipart forms

**Review by Vince-janv — COMMENTED** [review]
Some comments to address, but seems to be working as intended 👍

File: cg/clients/freshdesk/freshdesk_client.py (hunk 3 above)
Comment (ahdamin) [reply cg/clients/freshdesk/freshdesk_client.py:76]: Do you ignore `PylintW1203:logging-fstring-interpolation`: Use lazy % formatting in logging functions ?

File: cg/clients/freshdesk/freshdesk_client.py [hunk 7]
```diff
@@ -60,3 +58,27 @@ def _configure_retries(session: Session) -> None:
         )
         adapter = HTTPAdapter(max_retries=retry_strategy)
         session.mount("https://", adapter)
+
+    @handle_client_errors
+    def reply_to_ticket(
+        self, ticket_id: str, from_email: str, reply: dict, attachments: List[Path] = None
+    ) -> None:
```
[the remark this answers is not in the document]
Comment (ahdamin) [reply cg/clients/freshdesk/freshdesk_client.py]: done ✅

File: cg/clients/freshdesk/freshdesk_client.py (hunk 5 above)
Comment (ahdamin) [reply cg/clients/freshdesk/freshdesk_client.py]: ✅

File: cg/meta/orders/ticket_handler.py [hunk 8]
```diff
@@ -185,27 +190,25 @@ def replace_empty_string_with_none(cls, obj: Any) -> Any:
         return obj
 
     def connect_to_ticket(
-        self, order: OrderIn, user_name: str, user_mail: str, project: str, ticket_number: str
+        self, order: OrderIn, user_name: str, project: str, ticket_number: str
     ) -> None:
         """Appends a new order message to the ticket selected by the customer"""
-        LOG.info(f"Connecting order to ticket {ticket_number}")
+        LOG.info("Connecting order to ticket %s", ticket_number)
```
Comment (ChrOertlin) [inline cg/meta/orders/ticket_handler.py:196]: For all string building purposes we use f strings. Can you check the other instances throughout the PR and align it to that?
```suggestion
        LOG.info(f"Connecting order to ticket {ticket_number})
```

**Pull request merged**